From f3161f36f5f49ee995b90d6207d8b6d990355e0a Mon Sep 17 00:00:00 2001
From: Linus Torvalds <torvalds@linux-foundation.org>
Date: Thu, 23 Oct 2008 11:54:08 -0700
Subject: [PATCH] pcc-acpi: fix compile with new stricter ACPI types

This was another merge problem that wasn't a data conflict, but due to
independent changes in two branches that just didn't work together.

The pcc-acpi staging driver used

	acpi_driver_data(device) = hotkey;

to set driver data, but the ACPI merge made that invalid in commit
db89b4f0dbab837d0f3de2c3e9427a8d5393afa3 ("ACPI: catch calls of
acpi_driver_data on pointer of wrong type"), and now you're supposed to
just do.

	device->driver_data = hotkey;

instead.

Fix it up.

Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
---
 drivers/staging/pcc-acpi/pcc-acpi.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/pcc-acpi/pcc-acpi.c b/drivers/staging/pcc-acpi/pcc-acpi.c
index 7715c31..3b34a3e 100644
--- a/drivers/staging/pcc-acpi/pcc-acpi.c
+++ b/drivers/staging/pcc-acpi/pcc-acpi.c
@@ -1011,7 +1011,7 @@ static int __devinit acpi_pcc_hotkey_add(struct acpi_device *device)
 	hotkey->device   = device;
 	hotkey->handle   = device->handle;
 	hotkey->num_sifr = num_sifr;
-	acpi_driver_data(device) = hotkey;
+	device->driver_data = hotkey;
 	strcpy(acpi_device_name(device),  ACPI_PCC_DEVICE_NAME);
 	strcpy(acpi_device_class(device), ACPI_PCC_CLASS);
 
-- 
1.5.6.5

